File: C++/2021/ArrangingBooks.cpp
#include <iostream>
#include <algorithm>

using namespace std;

int main() {
    string shelf;
    cin >> shelf;

    int numl = count(shelf.begin(), shelf.end(), 'L');
    int numm = count(shelf.begin(), shelf.end(), 'M');
    int nums = count(shelf.begin(), shelf.end(), 'S');

    string l_section = shelf.substr(0, numl);
    string m_section = shelf.substr(numl, numm);
    string s_section = shelf.substr(numl + numm);

    int lm_section_displaced_max = max(count(l_section.begin(), l_section.end(), 'M'), count(m_section.begin(), m_section.end(), 'L'));
    int s_section_displaced = s_section.length() - count(s_section.begin(), s_section.end(), 'S');

    cout << lm_section_displaced_max + s_section_displaced << endl;
}

File: Python/2021/ArrangingBooks.py
# J4

shelf = input()

num_l, num_m, num_s = shelf.count("L"), shelf.count("M"), shelf.count("S")

l_section = shelf[:num_l]
m_section = shelf[num_l:num_m+num_l]
s_section = shelf[num_m+num_l:]

print(max(l_section.count("M"), m_section.count("L")) + (len(s_section) - s_section.count("S")))
